wfe/wfe.go

```diff
@@ -1150,6 +1158,31 @@ func (wfe *WebFrontEndImpl) Registration(ctx context.Context, logEvent *requestE
 		return
 	}
 
+	// If key rollover has been enabled people may try to POST a different key
+	// in the registration update body, if they do so and the key doesn't match
+	// the current registration key that we are going to overwrite the update key
+	// with below then throw an error informing them of the proper endpoint for
+	// key rollover.
+	if features.Enabled(features.AllowKeyRollover) && update.Key.Key != nil {
+		crk, err := x509.MarshalPKIXPublicKey(currReg.Key.Key)
+		if err != nil {
+			logEvent.AddError("unable to marshal registration key: %s", err)
+			wfe.sendError(response, logEvent, probs.ServerInternal("Unable to marshal registration key"), nil)
+			return
+		}
+		urk, err := x509.MarshalPKIXPublicKey(update.Key.Key)
+		if err != nil {
+			logEvent.AddError("unable to marshal registration key in POST body: %s", err)
+			wfe.sendError(response, logEvent, probs.Malformed("Unable to marshal registration key in POST body"), nil)
+			return
+		}
+		if bytes.Compare(urk, crk) != 0 {
+			msg := "Key can only be changed using the /acme/key-change endpoint"
```

Comment:
Can you use the `rolloverPath` for this error message? One less place to update if it changes down the road.